### PR TITLE
Chore: Refactor to use ES6 Classes (Part 2)

### PR DESCRIPTION
**Pull request by gyandeeps**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ x ] Other, please explain:

**What changes did you make? (Give an overview)**

Continue the work from #7846 

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -184,150 +184,152 @@ function getLocalConfig(thisConfig, directory) {
...
+         * Handle declared globals.
+         * For global variable foo, handle "foo:false" and "foo:true" to set
+         * whether global is writable.
+         * If user declares "foo", convert to "foo:false".
+         */
+        this.globals = (options.globals || []).reduce((globals, def) => {
+            const parts = def.split(":");
 
-        globals[parts[0]] = (parts.length > 1 && parts[1] === "true");
+            globals[ parts[ 0 ] ] = (parts.length > 1 && parts[ 1 ] === "true");
```
Comment (platinumazure) [inline lib/config.js]: FYI, not-an-aardvark has a chore PR open to set `array-bracket-spacing` to "never", so you may wish to either revert the spacing changes here or find his PR and object to it.

Comment (gyandeeps) [inline lib/config.js]: lol. I was actually looking into this since my webstrom is used to doing this.

Comment (platinumazure) [inline lib/config.js]: If only WebStorm could be configured to use a certain linting tool... 😉 LGTM now.

File: lib/cli-engine.js [hunk 2]
```diff
@@ -437,136 +437,133 @@ function getCacheFile(cacheFile, cwd) {
  * @param {CLIEngineOptions} options The options for this instance.
  * @constructor
  */
-function CLIEngine(options) {
+class CLIEngine {
```
Comment (nzakas) [inline lib/cli-engine.js]: Again, this would be a breaking change. We can't do it right now.

Comment (gyandeeps) [inline lib/cli-engine.js]: @nzakas done.

**Comment by gyandeeps** [comment]
@kaicataldo Done

**Pull request merged**